RockLib.Logging/Logger.cs

```diff
@@ -122,7 +130,13 @@ public enum ProcessingMode
         /// Log entries with a level lower than the value of this property are
         /// not logged by this logger.
         /// </remarks>
-        public LogLevel Level { get; }
+        public LogLevel Level
+        {
+            get
+            {
+                return _logLevelResolver?.GetLogLevel() ?? _level;
```

Comment:
It would be nice if the log level resolver could support asynchronous log level retrieval, but it's kind of more than it's worth. If the consumer has to use async to do this, then they'll just have to manage it themselves at this point.

There's tons of options for this - they could use a background service that runs asynchronously to pull the log level and share it with the resolver.. or they could manage executing an async method synchronously.. etc.